### PR TITLE
treewide: fix eval related to `with lib;` removal

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

fixup for #335603, #335618

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eclairevoyant** [comment]
I can't promise that this fixes _every_ problem but I caught what I saw

**Comment by philiptaron** [comment]
`lib.map` is net neutral because it's in the prelude.

File: nixos/modules/config/resolvconf.nix [hunk 1]
```diff
@@ -35,7 +35,7 @@ in
     (lib.mkRenamedOptionModule [ "networking" "dnsExtensionMechanism" ] [ "networking" "resolvconf" "dnsExtensionMechanism" ])
     (lib.mkRenamedOptionModule [ "networking" "extraResolvconfConf" ] [ "networking" "resolvconf" "extraConfig" ])
     (lib.mkRenamedOptionModule [ "networking" "resolvconfOptions" ] [ "networking" "resolvconf" "extraOptions" ])
-    (lib.mkRemovedOptionModule [ "networking" "resolvconf" "useHostResolvConf" ] "This option was never used for lib.anything lib.anyways")
+    (lib.mkRemovedOptionModule [ "networking" "resolvconf" "useHostResolvConf" ] "This option was never used for anything anyways")
```
Comment (philiptaron) [inline nixos/modules/config/resolvconf.nix:38]: Real issue, only affects diagnostics.

File: nixos/modules/services/continuous-integration/jenkins/default.nix [hunk 2]
```diff
@@ -205,13 +205,13 @@ in {
       preStart =
         let replacePlugins =
               lib.optionalString (cfg.plugins != null) (
-                let pluginCmds = lib.attrsets.lib.mapAttrsToList
+                let pluginCmds = lib.attrsets.mapAttrsToList
```
Comment (philiptaron) [inline nixos/modules/services/continuous-integration/jenkins/default.nix:208]: Real issue

File: nixos/modules/services/continuous-integration/jenkins/default.nix [hunk 3]
```diff
@@ -205,13 +205,13 @@ in {
...
                       (n: v: "cp ${v} ${cfg.home}/plugins/${n}.jpi")
                       cfg.plugins;
                 in ''
                   rm -r ${cfg.home}/plugins || true
                   mkdir -p ${cfg.home}/plugins
-                  ${lib.strings.lib.concatStringsSep "\n" pluginCmds}
+                  ${lib.strings.concatStringsSep "\n" pluginCmds}
```
Comment (philiptaron) [inline nixos/modules/services/continuous-integration/jenkins/default.nix:214]: Real issue

**Review by philiptaron — APPROVED** [review]
There are four real issues here, three that affect eval/functionality and one that affects diagnostics. The rest of the files can be reverted or committed, it doesn't really matter. 

I _prefer_ to revert, since `map` is shorter than `lib.map`, but I'll leave that up to you, @eclairevoyant.

**Comment by eclairevoyant** [comment]
> These symbols are in the Nix prelude, are available without any import, and also happen to be available from `lib`:

Thanks for pointing it out, I've removed those changes.

**Comment by anpin** [comment]
please add this as well 
```patch
diff --git a/nixos/modules/services/hardware/thinkfan.nix b/nixos/modules/services/hardware/thinkfan.nix
index 9dd4c5434211..9733fbe5aa15 100644
--- a/nixos/modules/services/hardware/thinkfan.nix
+++ b/nixos/modules/services/hardware/thinkfan.nix
@@ -12,7 +12,7 @@ let
       tuple = ts: lib.mkOptionType {
         name = "tuple";
         merge = lib.mergeOneOption;
-        check = xs: all id (zipListsWith (t: x: t.check x) ts xs);
+        check = xs: lib.all lib.id (lib.zipListsWith (t: x: t.check x) ts xs);
         description = "tuple of" + lib.concatMapStrings (t: " (${t.description})") ts;
       };
       level = ints.unsigned;
```

**Comment by eclairevoyant** [comment]
That's already fixed (#339084)

**Review by philiptaron — APPROVED** [review]

**Pull request merged**